tests/Behat/Context/Ui/Admin/ManagingInvoicesContext.php

```diff
@@ -293,4 +301,35 @@ public function shouldBeNotifiedThatEmailWasSentSuccessfully(): void
             NotificationType::success()
         );
     }
+
+    /**
+     * @Then it should have :quantity :name shipment with unit price :unitPrice, net value :netValue, tax total :taxTotal and total :total in :currencyCode currency
+     */
+    public function itShouldHaveShipmentWithUnitPriceNetValueTaxTotalAndTotalInCurrency(
+        int $quantity,
+        string $name,
+        string $unitPrice,
+        string $netValue,
+        string $taxTotal,
+        string $total,
+        string $currencyCode
+    ): void {
+        Assert::true($this->showPage->hasItemWithData($name, $unitPrice, $quantity, $taxTotal, $total, $currencyCode, $netValue));
+    }
+
+    /**
+     * @Then it should have :quantity :name item with unit price :unitPrice, net value :netValue, tax total :taxTotal and total :total in :currencyCode currency
+     * @Then it should have :quantity :name items with unit price :unitPrice, net value :netValue, tax total :taxTotal and total :total in :currencyCode currency
+     */
```

Comment:
Wouldn't here this funny structure with `?:s` ( I don't remember it exactly :D ) work?

Reply:
You are right, I would use the structure with optional `s`, but then I would have to change the whole definition to a regexp, which in my opinion is less readable with so many arguments

Reply:
👌 I will stay with Team readable then 😄